src/math/numbers.c

```diff
@@ -129,3 +129,27 @@ int norm_int32(int32_t val)
 	}
 	return s;
 }
+
+/**
+ * Basic CRC-32 implementation based on crc32b from:
+ * http://www.hackersdelight.org/hdcodetxt/crc.c.txt
+ * 0xEDB88320 is the reversed polynomial representation
+ */
+uint32_t crc32(const void *data, uint32_t bytes)
```

Comment:
the CRC should pass the polynomial as parameter to be generic. Also why do we need a CRC for hotwording?

Reply:
Verification of params passed via packeted large IPCs

Reply:
Can the usage of the licence be pasted in this comment too in order to make it clear about the use.
Shouldn't the polynomial parameter be in the ABI definition too since we will have to run the CRC twice (once on host and once if FW).

Reply:
Any comment on the polynomial parameter within ABI ?

Reply:
Sorry, mind elaborating?

Reply:
Wont the same polynomial parameter be needed by the sender (to generate the CRC) and by the recipient to validate ? Sorry I last looked at CRC's at university which was a long time ago...

Reply:
Not necessarily, it just has to be the same CRC type on both ends. This is just standard CRC-32, which can use a number of different polynomial representations, all depends on the implementation.

Reply:
@slawblauciak we need a textbook CRC that can be licensed BSD/MIT.

Reply:
Well, this is basically it. I removed the reference to the previous source, since it's irrelevant. All implementations that don't use a lookup table look pretty much identical **and** they explicitly state that they're public domain.